sunpy/net/helio/hec.py

```diff
@@ -173,8 +173,8 @@ def search(self, *args, **kwargs):
                                                     MAXRECORDS=max_records)
         results = votable_handler(etree.tostring(results))
         table = HECResponse(results.to_table(), client=self)
-        if len(table) == max_records == 500:
-            warn_user("Number of results is the same as the default `max_records` of 500. "
+        if len(table) == max_records:
+            warn_user("Number of results is the same as current limit. "
```

Comment:
So now we just need a unit test that hits this message and I think this PR will be good to go.

Reply:
Is it possible for `assert len(res) == 10` not to trigger the warning?

Reply:
It doesn't seem to?

Reply:
I think we can hits the warning like this:
```python
@pytest.mark.remote_data
def test_client_search(client):
    start = '2005/01/03'
    end = '2005/12/03'
    table_name = 'rhessi_hxr_flare'
    with pytest.warns(SunpyUserWarning, match="Number of results is the same as current limit. "
                                         "It is possible your query has been truncated. "
                       "If you want to change this, set `a.helio.MaxRecords` to a higher value."):
        res = client.search(a.Time(start, end), a.helio.TableName(table_name), a.helio.MaxRecords(10))
    assert len(res) == 10
``` 

But, should we keep `assert len(res) == 10`?